src/core/Config.cpp

```diff
@@ -624,11 +624,13 @@ void Config::createTempFileInstance()
     if (m_instance) {
         delete m_instance;
     }
-    auto* tmpFile = new QTemporaryFile();
-    bool openResult = tmpFile->open();
-    Q_ASSERT(openResult);
-    Q_UNUSED(openResult);
-    m_instance = new Config(tmpFile->fileName(), "", qApp);
+    auto tmpFileName = QString("%1/%2_settings.XXXXXX").arg(QDir::tempPath(), QCoreApplication::applicationName());
+    auto tmpFile = new QTemporaryFile(tmpFileName);
+    if (!tmpFile->open()) {
+        Q_ASSERT_X(false, __func__, "Failed to create temporary config settings file");
+    }
+    tmpFile->close();
+    m_instance = new Config(tmpFileName, "", qApp);
     tmpFile->setParent(m_instance);
```

Comment:
I was wondering what this function is even for. It looks like it's only used in tests. Can we give it a better name, so it's clear what this is for?

Reply:
Yes sure, we might be able to remove this entirely since we call another public function anyway.

Reply:
OK figured this out, we are smart cause we created `TemporaryFile` that gets around the QTemporaryFile nonsense. Forgot about that!